src/transformers/integrations/awq.py

```diff
@@ -145,6 +147,17 @@ def replace_with_awq_linear(
 
                 # Force requires grad to False to avoid unexpected errors
                 model._modules[name].requires_grad_(False)
+        if isinstance(module, nn.GELU) and name not in modules_to_not_convert:
+            # get the module just before applying act
+            # the easiest way is just checking if the mlp have these layers.
+            layers = ["fc_in", "dense_h_to_4h", "up_proj", "c_fc"]
```

Comment:
This seems super brittle - let's do this in a more generalized way. In particular, this doesn't check these layers are used before applying the activation

Reply:
Sounds good ! I decided to check for the model type and retrieve the layer before the activation.